metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -326,7 +330,28 @@ func createWatcher(
 		if isNamespaced(resourceName) {
 			options.Namespace = namespace
 		}
-		watcher, err := kubernetes.NewNamedWatcher(resourceName, client, resource, options, nil)
+		var (
+			watcher kubernetes.Watcher
+			err     error
+		)
+		switch resource.(type) {
+		// use a metadata informer for ReplicaSets, as we only need their metadata
+		case *kubernetes.ReplicaSet:
+			if err != nil {
+				return false, err
+			}
+			watcher, err = kubernetes.NewNamedMetadataWatcher(
+				"resource_metadata_enricher_rs",
+				client,
+				metadataClient,
+				schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "replicasets"},
+				options,
+				nil,
+				metadata.RemoveUnnecessaryReplicaSetData,
+			)
```

Comment:
That looks to be the other way around:
```suggestion
			watcher, err = kubernetes.NewNamedMetadataWatcher(
				"resource_metadata_enricher_rs",
				client,
				metadataClient,
				schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "replicasets"},
				options,
				nil,
				metadata.RemoveUnnecessaryReplicaSetData,
			)
			if err != nil {
				return false, err
			}
```

Reply:
If that's really the case, adding some some extra information/context to the error would be nice, if it makes sense in this context.

Reply:
This error is actually checked after the switch statement. I'll make it a bit nicer.